### ISSUE TITLE
Config Contexts with Racks or Locations

### ISSUE THREAD

**Issue by Joshi425** [netbox-community/netbox#7364]
### NetBox version

v3.0.2

### Feature type

Data model extension

### Proposed functionality

make config contexts work with locations and racks 

### Use case

we use a spine leaf setup in our dc and some maschines need to be aware of the leaf they are in. To accomplish this I wanted to create locations with the racks as children or just assign a config context to the racks in each leaf. But neither is supported yet

https://netbox.readthedocs.io/en/stable/models/extras/configcontext/

**Comment by kkthxbye-code** [comment]
Going out on a limb, what is the reasoning for not having config context on all primary, organizational and nested group models?

https://netbox.readthedocs.io/en/stable/development/models/

It seems a little arbitrary maybe, as confix contexts are avaliable for at least one model in each catagory.

**Comment by jeremystretch** [comment]
Associating config contexts with individual racks has been discussed previously, under #4158 and #4699, and we have decided not to implement it for the reasons cited there.

**Comment by Joshi425** [comment]
okay I see I can also make this connection with IP Ranges. Is there a way to assign a config context to IP Ranges or how can I get this information inherited down?

**Comment by Joshi425** [comment]
@jeremystretch sorry to mention you but do you have a suggestion or point me to a way? I want to do it the right way.

**Comment by jeremystretch** [comment]
No, there is no relation between config contexts and IP ranges. That wouldn't make any sense. If you need help with a specific solution, please start a new [discussion](https://github.com/netbox-community/netbox/discussions/new).
